src/java.base/share/classes/jdk/internal/logger/LoggerFinderLoader.java

```diff
@@ -67,18 +73,32 @@ private LoggerFinderLoader() {
 
 
     // Return the loaded LoggerFinder, or load it if not already loaded.
+    static volatile Thread loadingThread;
     private static System.LoggerFinder service() {
         if (service != null) return service;
         synchronized(lock) {
             if (service != null) return service;
-            service = loadLoggerFinder();
+            Thread currentThread = Thread.currentThread();
+            if (loadingThread == currentThread) {
+                return new TemporaryLoggerFinder();
```

Comment:
```suggestion
                // recursive ttempt to load the backend while loading the backend
                // use a temporary logger finder that returns special BootsrtapLoggers
                // which will wait until loading is finished
                return new TemporaryLoggerFinder();
```

Reply:
We could create a singleton instance of TemporaryLoggerFinder in the TemporaryLoggerFinder class and return that.
```suggestion
                return new TemporaryLoggerFinder.INSTANCE;
```

Reply:
> We could create a singleton instance of TemporaryLoggerFinder in the TemporaryLoggerFinder class and return that.

I think the disadvantage is that this singleton instance will never be GCed (till `LoggerFinderLoader` class itself is unloaded) unlike just using a new instance at the call site. Having said that, not being GCed doesn't make a huge difference given what this singleton instance does or holds on to.